torch/_inductor/fx_passes/pre_grad.py

```diff
@@ -22,12 +22,14 @@
 
 log = logging.getLogger(__name__)
 
+normalize_cat_pass = PatternMatcherPass(prevent_match_across_mutations=True)
 normalize_split_pass = PatternMatcherPass(prevent_match_across_mutations=True)
 merge_splits_pass = PatternMatcherPass(prevent_match_across_mutations=True)
 split_cat_pass = PatternMatcherPass(prevent_match_across_mutations=True)
 unbind_stack_pass = PatternMatcherPass(prevent_match_across_mutations=True)
 
 pattern_matcher_passes: List[PatternMatcherPass] = [
+    normalize_cat_pass,
```

Comment:
Maybe we can do this normalization in `normalize_split_pass`, as that will be more efficient? (We can rename it to `normalize_pass`)